File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        slow = fast = head
        firstCycle = True
        
        while fast and fast.next:
            if slow == fast and not firstCycle:
                return True
            slow = slow.next
            fast = fast.next.next
            firstCycle = False
        return False

File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
const hasCycle = head => {
  while (head) {
    if (head.val === 'done') return true;

    head.val = 'done';
    head= head.next;
  }

  return false;
};
  